.github/workflows/quality-checks.yml

```diff
@@ -0,0 +1,40 @@
+name: Run gofmt & unit tests
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - '**/**.go'
+  pull_request:
+    paths:
+      - '**/**.go'
+      - 'Makefile'
+      - '.github/workflows/quality-checks.yml'
+
+jobs:
+  quality-checks:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v3
+    
+    - name: Set up Go
+      uses: actions/setup-go@v3
+      with:
+        go-version: 1.22
+
+    - name: Install dependencies
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y libnlopt-dev
```

Comment:
ugh, our `go.viam.com/test` dep uses nlopt

Reply:
.... do we need to use our own test package? This seems ridiculous.

Reply:
Oh that's why I needed that haha...

Reply:
but also how is it gettign nlopt? https://github.com/viamrobotics/test, this seems like it should not need it at all